### PR TITLE
Add support for npm package installations to Azure DevTest Labs artifacts

### PR DESCRIPTION
**Pull request by d3r3kk**
# ALM Rangers Artifact: NPM #
## Package Management for Node.js on Linux ##

Exposes the nodejs package management system for Linux systems to the Azure DevTest Labs users.

Now users can install whatever software they wish via npm in a way that is natural to people familiar with Node.js.

Please see README.md for list of supported distributions and usage.

> Note: This does not install NPM as part of its execution. This is due to the vast ways that users will want to make use of to get the nodejs/npm version of their choice onto the Linux distribution of their choice. The prescribed manner that people will use this artifact is part of a chain such as: Bash artifact to setup the PPA for the desired version of node on the desired Linux distribution + Apt or Yum artifact to install nodejs/npm + this artifact to install npm packages.

### CODE REVIEW

File: Artifacts/linux-npm-package/Artifactfile.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+    "$schema": "https://raw.githubusercontent.com/Azure/azure-devtestlab/master/schemas/2015-01-01/dtlArtifacts.json",
+    "title": "NPM Package",
+    "description": "Install NPM packages on linux using the Node Package Manager.",
```
Comment (drwill-ms) [inline Artifacts/linux-npm-package/Artifactfile.json]: Linux (capital L)

File: Artifacts/linux-npm-package/Artifactfile.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            "type": "string",
+            "displayName": "Install Location",
+            "defaultValue": "/var/tmp",
+            "allowEmpty": false,
+            "description": "The local path to install the package to or the keyword 'global' to install the package(s) globally instead."
+        },
+        "packages": {
+            "type": "string",
+            "displayName": "Packages to Install",
+            "description": "The package names you wish to have installed via NPM, separated by commas. The packages will be installed globally. If NPM is not yet on the system, it also will be installed. You can find NPM package names in the index found at https://www.npmjs.com. All npm install commands are supported, however if a git/bitbucket/gist scope is given then you must have installed the git artifact prior to this npm package artifact."
```
Comment (drwill-ms) [inline Artifacts/linux-npm-package/Artifactfile.json]: Why does this say they'll be installed globally but the previous parameter provides an option to install globally or to a specific location?

Comment (drwill-ms) [inline Artifacts/linux-npm-package/Artifactfile.json]: NPM (all caps)

File: Artifacts/linux-npm-package/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+The script is intended for use in the Azure DevTest Labs artifact system, and the parameters for the artifact are fed directly
+into the script. However, you can run it from any bash shell using the following format:
+
+        bash> .\linux-npm-package.sh -packages (PACKAGE-LIST) --install-to (global|/path/to) --options [ADDITIONAL-OPTIONS]
+
+**Where:**
+
+*packages*
+
+The names of the packages to install, separated by spaces. Each package declaration follows the allowable declarations in npm
```
Comment (drwill-ms) [inline Artifacts/linux-npm-package/README.md]: The artifact said delimit by commas. Does the script need just spaces instead?

File: Artifacts/linux-npm-package/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+*install-to*
+
+Defaults to global, the 'install-to' parameter tells npm where to install the named packages. This can be the keyword 'global' 
+or left blank (default is global) to install to the global location (typically /usr/lib/node_modules on a Linux system). Or the
+value can be any valid path (ie. /var/tmp/dev) to install the npm packages into.
+
+*options*
+
+Any other valid arguments that the npm-install command accepts. Since these additional parameters allow for relatively infinite permutations
+they will likely be susceptible to failure on the images supplied by the Azure DevTest Labs. Advanced users who wish to push the limiatations
```
Comment (drwill-ms) [inline Artifacts/linux-npm-package/README.md]: limitations (misspelled)

**Comment by d3r3kk** [comment]
I've updated the PR with the requested change. Is there something else?

**Pull request merged**